File: 프로그래머스/1/12901. 2016년/2016년.js
function solution(a, b) {
    var answer = 0;
    var date = {1:31, 2:29, 3:31, 4:30, 5:31, 6:30, 7:31, 8:31, 9:30, 10:31, 11:30, 12:31};
    var day = ["THU","FRI","SAT","SUN","MON","TUE","WED"];
    while(a!==1){
        a--;
        answer += date[a];
    }
    answer += b;
    return day[answer%7];
}

File: 프로그래머스/1/12901. 2016년/2016년.java
import java.time.LocalDate;
import java.time.temporal.ChronoUnit;

class Solution {
    public String solution(int a, int b) {
        String answer = "";
        String[] arr = {"FRI", "SAT", "SUN", "MON", "TUE", "WED", "THU"};

        LocalDate start = LocalDate.of(2016, 1, 1);
        LocalDate end = LocalDate.of(2016, a, b);

        int days = (int) ChronoUnit.DAYS.between(start, end);

        answer = arr[days % arr.length];

        return answer;
    }
}